models/demos/wormhole/distilbert/tt/ttnn_optimized_distilbert.py

```diff
@@ -0,0 +1,419 @@
+
+
+def transformer(
+    config,
+    x,
+    attention_mask=None,
+    head_mask=None,
+    output_attentions: bool = False,
+    output_hidden_states: bool = False,
+    base_address=None,
+    parameters=None,
+    device=None,
+    min_val_tensor=None,
+    negative_val_tensor=None,
+    mesh_mapper=None,
+):
+    all_hidden_states = () if output_hidden_states else None
+    all_attentions = () if output_attentions else None
+    hidden_state = x
+
+    for params in parameters.layer:
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_state,)
+
+        layer_outputs = transformer_block(
+            config=config,
+            x=hidden_state,
+            attention_mask=attention_mask,
+            head_mask=None,
+            output_attentions=output_attentions,
+            base_address=f"{base_address}.layer",
+            parameters=params,
+            device=device,
+            min_val_tensor=min_val_tensor,
+            negative_val_tensor=negative_val_tensor,
+        )
+        hidden_state = layer_outputs
+
+    return hidden_state
+
+
+def distilbert(
+    config,
+    input_ids=None,
+    attention_mask=None,
+    head_mask=None,
+    inputs_embeds=None,
+    output_attentions=None,
+    output_hidden_states=None,
+    position_ids=None,
+    min_val_tensor=None,
+    negative_val_tensor=None,
+    *,
+    base_address,
+    parameters,
+    device,
+    mesh_mapper=None,
+    ip_mesh_mapper=None,
+):
+    output_attentions = output_attentions if output_attentions is not None else config.output_attentions
+    output_hidden_states = output_hidden_states if output_hidden_states is not None else config.output_hidden_states
+    if input_ids is not None and inputs_embeds is not None:
+        raise ValueError("You cannot specify both input_ids and inputs_embeds at the same time")
+    elif input_ids is not None:
+        input_shape = input_ids.shape
+    elif inputs_embeds is not None:
+        input_shape = inputs_embeds.size()[:-1]
+    else:
+        raise ValueError("You have to specify either input_ids or inputs_embeds")
+
+    head_mask = get_head_mask(head_mask, config.num_hidden_layers)
+
+    if input_ids is not None:
+        word_embeddings = ttnn.embedding(
+            input_ids,
+            parameters.distilbert.embeddings.word_embeddings.weight,
+            layout=ttnn.TILE_LAYOUT,
+            memory_config=ttnn.DRAM_MEMORY_CONFIG,
+        )
+    seq_length = word_embeddings.shape[1]
+
+    if position_ids is not None:
+        position_ids = position_ids[:, :seq_length]
+    else:
+        # This part is not executed
+        position_ids = ttnn.arange(0, seq_length, step=1, device=device, memory_config=ttnn.L1_MEMORY_CONFIG)
+        position_ids = ttnn.reshape(position_ids, (1, input_ids.shape[-1]))
+        # to_layout fails to convert dtype.
+        # Issue: https://github.com/tenstorrent/tt-metal/issues/9440
+        position_ids = ttnn.to_torch(position_ids)
```

Comment:
Moving to host and back is slow. I appreciate this is a workaround but if the only goal is to change the type then you could use https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/clone.html instead and keep it on device.